python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -542,42 +584,339 @@ def asset_key_for_input(self, input_name: str) -> AssetKey:
+
         Raises an error if either of the following are true:
         - The output asset has no partitioning.
         - The output asset is not partitioned with a TimeWindowPartitionsDefinition or a
         MultiPartitionsDefinition with one time-partitioned dimension.
+
+        Args:
+            output_name (str): The name of the output produced by the asset to get the time window for.
+                If not provided for assets defined with the `@asset` decorator, the name of the output
+                will be automatically provided.
+
+        Examples:
+            .. code-block:: python
+
+                partitions_def = DailyPartitionsDefinition("2023-08-20")
+
+                @asset(
+                    partitions_def=partitions_def
+                )
+                def an_asset(context: AssetExecutionContext):
+                    context.log.info(context.asset_partitions_time_window_for_output())
+
+                    return 1
+
+                # materializing the 2023-08-21 partition of this asset will log:
+                #   TimeWindow("2023-08-21", "2023-08-22")
+
+                @multi_asset(
+                    outs={
+                        "first_asset": AssetOut(key=["my_assets", "first_asset"]),
+                        "second_asset": AssetOut(key=["my_assets", "second_asset"])
+                    }
+                    partitions_def=partitions_def,
+                )
+                def a_multi_asset(context: AssetExecutionContext):
+                    context.log.info(context.asset_partitions_time_window_for_output("first_asset"))
+                    context.log.info(context.asset_partitions_time_window_for_output("second_asset"))
+
+                    return 1, 2
+
+
+                # materializing the 2023-08-21 partition of this asset will log:
+                #   TimeWindow("2023-08-21", "2023-08-22")
+                #   TimeWindow("2023-08-21", "2023-08-22")
+
+
+                @asset(
+                    partitions_def=partitions_def,
+                    ins={
+                        "self_dependent_asset": AssetIn(partition_mapping=TimeWindowPartitionMapping(start_offset=-1, end_offset=-1))
+                    }
+                )
+                def self_dependent_asset(context: AssetExecutionContext, self_dependent_asset):
+                    context.log.info(context.asset_partitions_time_window_for_output())
+
+                # materializing the 2023-08-21 partition of this asset will log:
+                #   TimeWindow("2023-08-21", "2023-08-22")
+
         """
         return self._step_execution_context.asset_partitions_time_window_for_output(output_name)
 
     @public
     def asset_partition_key_range_for_output(
         self, output_name: str = "result"
     ) -> PartitionKeyRange:
-        """Return the PartitionKeyRange for the corresponding output. Errors if not present."""
+        """Return the PartitionKeyRange for the corresponding output. Errors if not present.
```

Comment:
I'm not sure what "Errors if not present" means here. I'd like to expand on what conditions cause an error, is it just if the output_name doesnt exist?

cc @clairelin135

Reply:
I think it should only error if the run is not partitioned (isn't a single-partition run or partition range run)